drafts/25-xxx-specifications.txt

```diff
@@ -296,25 +296,31 @@ di07. The non-directive directive
 7 Predefined macros
 ===================
 
-pm01. Any macro name predefined by the implementation shall begin with a
+pm00. Macro names beginning with a leading underscore followed by an uppercase
+      letter or a second underscore are reserved to the processor. Reserved
+      names shall not be the subject of a #define or #undef preprocessing
+      directive within a program unit.
+      [Note: Any valid macro name matching the regular expression `/^_[A-Z_].*/`
```

Comment:
Two things: I would rather we call out the full regular expression, not leave it to the reader to piece together `valid macro name` and the use of `.*`.

Also, since `^` means beginning-of-line in regex-speak, I would prefer to omit it. We're matching a macro name, not a line.

```suggestion
      [Note: Any macro name matching the regular expression `/_[A-Z][A-Za-z0-9_]*/`
```

Reply:
@gklimowicz Explicit is good, agreed on expanding out the macro.

Re: `^`, according to POSIX anyway, [`^` anchors the expression to the beginning of the string](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap09.html#tag_09_03_08) rather than the beginning of the line. (To have `^` match the beginning of any line, you have to add the `m` modifier in _e.g._ PCRE.)

Reply:
by the same argument, I guess the "correct" regex is actually `/_[A-Z][A-Za-z0-9_]*$/` since
> The <dollar-sign> shall anchor the expression (or optionally subexpression) to the end of the string being matched

Reply:
I'm good with both `^` and `$` being used in the regex.